### PR TITLE
adding falcon 40b-instruct

### PR DESCRIPTION
**Pull request by choudharykartik**
save to blob run link: https://ml.azure.com/experiments/id/0b3a7ac2-c071-44b6-b171-21ac17f51fb4/runs/good_zebra_n7gtdjr1sj?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourcegroups/2424-rg/workspaces/model-validations-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

evaluation run link: https://ml.azure.com/runs/musing_parsnip_mytvvnn71d?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourcegroups/2424-rg/workspaces/model-validations-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

import run link: https://ml.azure.com/experiments/id/4af4f931-e5c4-406d-a791-e84909c0a6ab/runs/sleepy_sock_1g5rq6mgbd?wsid=%2Fsubscriptions%2Fb17253fa-f327-42d6-9686-f3e553e24763%2FresourceGroups%2F2424-rg%2Fproviders%2FMicrosoft.MachineLearningServices%2Fworkspaces%2Fmodel-validations-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Endpoint response: 
image

conda.yaml:

```channels:
- conda-forge
dependencies:
- python=3.8.17
- pip<=23.2.1
- pip:
  - mlflow==2.3.1
  - cloudpickle==2.2.1
  - jsonpickle==3.0.2
  - mlflow-skinny==2.3.1
  - azureml-core==1.52.0
  - azureml-mlflow==1.52.0
  - azureml-metrics==0.0.24
  - scikit-learn==1.1.0
  - cryptography==41.0.2
  - python-dateutil==2.8.2
  - datasets==2.14.4
  - soundfile==0.12.1
  - librosa==0.10.0.post2
  - diffusers==0.19.3
  - sentencepiece==0.1.99
  - transformers==4.31.0
  - torch==2.0.0
  - accelerate==0.21.0
  - Pillow==9.4.0
  - einops
  - azureml-evaluate-mlflow==0.0.24
- pycocotools=2.0.4
name: mlflow-env```

MLModel:

```flavors:
  hftransformersv2:
    code: null
    config_hf_load_kwargs:
      trust_remote_code: true
    hf_config_class: AutoConfig
    hf_pretrained_class: AutoModelForCausalLM
    hf_tokenizer_class: AutoTokenizer
    huggingface_id: tiiuae/falcon-40b-instruct
    model_data: data
    model_hf_load_args:
      trust_remote_code: true
    pytorch_version: 2.0.0+cu117
    task_type: text-generation
    tokenizer_hf_load_kwargs:
      trust_remote_code: true
    transformers_version: 4.31.0
  python_function:
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.17
mlflow_version: 2.3.1
model_uuid: 5c3884d746a240f8976a0d20416477e5
utc_time_created: '2023-09-22 11:06:32.341334'
```

### CODE REVIEW

**Comment by novaturient95** [comment]
Please add conda.yaml and MLmodel file in description under code section

**Comment by novaturient95** [comment]
Few points for clarifications:
1. Import model registered model into ws, which is used for save to blobstorage
2. Separate ft and eval is using model from preview-test1. Could you please make sure same model is used across?

File: assets/models/system/tiiuae-falcon-40b-instruct/description.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+Hardware: Trained on AWS SageMaker using 64 A100 40GB GPUs in P4d instances
+Software: Utilizes a custom distributed training codebase called Gigatron
+
+# Recommendations and Limitations:
+
+Falcon-40B-Instruct may carry biases commonly found online due to its training data.
+Users are advised to implement guardrails and take precautions for production use.
+It's mostly suited for English and French and may not generalize well to other languages.
+
+> The above summary was generated using ChatGPT. Review the <a href="https://huggingface.co/tiiuae/falcon-40b" target="_blank">original model card</a> to understand the data used to train the model, evaluation metrics, license, intended uses, limitations and bias before using the model.
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-40b-instruct/description.md:20]: update model reference

File: assets/models/system/tiiuae-falcon-40b-instruct/description.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+# Inference samples
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.sh</a>
+Batch |<a href="https://aka.ms/azureml-infer-batch-sdk-text-generation" target="_blank">text-generation-batch-endpoint.ipynb</a>| coming soon
+
+### Model Evaluation Sample
+
+Task| Use case| Dataset| Python sample (Notebook)| CLI with YAML
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-40b-instruct/description.md:85]: have we validated evaluation with notebook mentioned here?

File: assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  evaluation_compute_allow_list:
+  - Standard_NC24rs_v3
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  inference_compute_allow_list:
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  model_specific_defaults:
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml:25]: please remove ft specific settings

**Comment by choudharykartik** [comment]
> Please add conda.yaml and MLmodel file in description under code section

Done

File: assets/models/system/tiiuae-falcon-40b-instruct/description.md (hunk 1 above)
Comment (choudharykartik) [reply assets/models/system/tiiuae-falcon-40b-instruct/description.md:20]: Done

File: assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml (hunk 3 above)
Comment (choudharykartik) [reply assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml:25]: Removed

File: assets/models/system/tiiuae-falcon-40b-instruct/model.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+path:
+  container_name: models
+  container_path: huggingface/tiiuae-falcon-40b/1695882007/mlflow_model_folder
```
[the remark this answers is not in the document]
Comment (choudharykartik) [reply assets/models/system/tiiuae-falcon-40b-instruct/model.yaml]: Updated, Thanks

**Comment by choudharykartik** [comment]
> Please make sure all information provided is correct

Yes, re-verified and updated the details.

File: assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  Preview: ''
+  license: apache-2.0
+  task: text-generation
+  author: tiiuae
+  evaluation_compute_allow_list:
+  - Standard_NC24rs_v3
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  inference_compute_allow_list:
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml:21]: inference on nc24 had failed?

**Comment by novaturient95** [comment]
rest all LGTM

File: assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml (hunk 5 above)
Comment (choudharykartik) [reply assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml:21]: Deployment was successful but inference was not working on NC24 due to timeout.

Comment (choudharykartik) [reply assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml:21]: yes request timeout issue with NC24 compute
image

File: assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: tiiuae-falcon-40b-instruct
+path: ./
+properties:
+  SHA: ca78eac0ed45bf64445ff0687fabba1598daebf3
+  datasets: tiiuae/falcon-refinedweb
+  languages: en
+  evaluation-recommended-sku: Standard_NC24rs_v3
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml:8]: Please add all computes as comma separated string

Comment (choudharykartik) [reply assets/models/system/tiiuae-falcon-40b-instruct/spec.yaml]: Done

**Pull request merged**